examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,239 @@
+use std::io::stdout;
+/// A Ratatui example that demonstrates the Elm architecture with a basic list - detail
+/// application.
+///
+/// This example runs with the Ratatui library code in the branch that you are currently
+/// reading. See the [`latest`] branch for the code which works with the most recent Ratatui
+/// release.
+///
+/// [`latest`]: https://github.com/ratatui/ratatui/tree/latest
+use std::time::Duration;
+
+use crossterm::event::{
+    self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, MouseEventKind,
+};
+use crossterm::ExecutableCommand;
+use ratatui::layout::{Constraint, Layout};
+use ratatui::style::{Modifier, Style};
+use ratatui::text::Line;
+use ratatui::widgets::{Block, Borders, Padding, Paragraph, Row, Table, TableState};
+use ratatui::Frame;
+
+static INFO_TEXT: &str = " (Esc/q) quit | (↑) move up | (↓) move down ";
+
+/// Application data model and state
+#[derive(Debug, Default)]
+struct Model {
```

Comment:
Would it be worth calling this AppModel - given the comment, this would seem to be aligned?